File: Angular/cookingdaddy/src/app/shared/ingredient/ingredient.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { Ingredient } from './ingredient';
import { Category } from './category';
import { Quality } from './quality';
import { Flavor } from './flavor';
import { UrlService } from '../url.service';

@Injectable({
  providedIn: 'root'
})
export class IngredientService {
  private appUrl = this.urlService.getUrl();
  private headers = new HttpHeaders({ 'Content-Type': 'application/json' });
  constructor( private urlService: UrlService, private http: HttpClient ) { }

  public getIngredients(): Observable<Ingredient[]> {
    return this.http.get( this.appUrl + '/ingredient', { withCredentials: true }).pipe( map( resp => resp as Ingredient[] ));
  }
  public getQualities(): Observable<Quality[]> {
    return this.http.get( this.appUrl + '/quality', { withCredentials: true }).pipe( map( resp => resp as Quality[] ));
  }
  public getFlavors(): Observable<Flavor[]> {
    return this.http.get( this.appUrl + '/flavor', { withCredentials: true }).pipe( map( resp => resp as Flavor[] ));
  }
  public getCategories(): Observable<Category[]> {
    return this.http.get( this.appUrl + '/category', { withCredentials: true }).pipe( map( resp => resp as Category[] ));
  }
  public getIngredient( id: number ) {
    const url = this.appUrl + '/ingredient/' + id;
    return this.http.get( url, { withCredentials: true }).pipe( map( resp => resp as Ingredient ));
  }
  public getCategory( id: number ) {
    const url = this.appUrl + '/category/' + id;
    return this.http.get( url, { withCredentials: true }).pipe( map( resp => resp as Category ));
  }
  public getQuality( id: number ) {
    const url = this.appUrl + '/quality/' + id;
    return this.http.get( url, { withCredentials: true }).pipe( map( resp => resp as Quality ));
  }
  public getFlavor( id: number ) {
    const url = this.appUrl + '/flavor/' + id;
    return this.http.get( url, { withCredentials: true }).pipe( map( resp => resp as Flavor ));
  }
  public addIngredient( ingredient: Ingredient ) {
    const body = JSON.stringify( ingredient );
    return this.http.post( this.appUrl + '/ingredient', body, { headers: this.headers, withCredentials: true }).pipe(
      map( resp => resp as Ingredient )
    );
  }
  public addCategory( category: Category ) {
    const body = JSON.stringify( category );
    return this.http.post( this.appUrl + '/category', body, { headers: this.headers, withCredentials: true }).pipe(
      map( resp => resp as Category )
    );
  }
  public addQuality( quality: Quality ) {
    const body = JSON.stringify( quality );
    return this.http.post( this.appUrl + '/quality', body, { headers: this.headers, withCredentials: true }).pipe(
      map( resp => resp as Quality )
    );
  }
  public addFlavor( flavor: Flavor ) {
    const body = JSON.stringify( flavor );
    return this.http.post( this.appUrl + '/flavor', body, { headers: this.headers, withCredentials: true }).pipe(
      map( resp => resp as Flavor )
    );
  }
  public updateIngredient( ingredient: Ingredient ): Observable<Ingredient> {
    const url = this.appUrl + '/ingredients/' + ingredient.id;
    const body = JSON.stringify( ingredient );
    return this.http.put(url, body, { headers: this.headers, withCredentials: true }).pipe(
      map( resp => resp as Ingredient )
    );
  }
}

File: CookingDaddy/src/main/java/com/revature/service/IngredientService.java
package com.revature.service;

import java.util.Set;

import com.revature.beans.Category;
import com.revature.beans.Flavor;
import com.revature.beans.Ingredient;
import com.revature.beans.PersonIngredient;
import com.revature.beans.Quality;
import com.revature.beans.Season;

public interface IngredientService {
	
	public Integer addIngredient(Ingredient ingredient);
	public Set<Ingredient> getIngredients();
	public Ingredient getIngredient(Integer id);
	public Ingredient updateIngredient(Ingredient ingredient);

	public Category getCategory(Integer id);
	public Set<Category> getCategories();
	public Integer addCategory(Category category);
	public Category updateCategory(Category category);
	
	public Flavor getFlavor(Integer id);
	public Set<Flavor> getFlavors();
	public Integer addFlavor(Flavor flavor);
	public Flavor updateFlavor(Flavor flavor);
	
	public Quality getQuality(Integer id);
	public Set<Quality> getQualities();
	public Integer addQuality(Quality quality);
	public Quality updateQuality(Quality quality);
	
	public Season getSeason(Integer id);
	public Set<Season> getSeasons();
	public Integer addSeason(Season season);
	public Season updateSeason(Season season);

	public Integer addPersonIngredient(PersonIngredient ping);
	public PersonIngredient updatePersonIngredient(PersonIngredient ping);
	public void deletePersonIngredient(PersonIngredient ping);
	
	public Set<PersonIngredient> getStarterIngredients(Integer personID);
}
